server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -724,9 +724,17 @@ private AllocateUnassignedDecision explainUnassignedShardAllocation(ShardRouting
 
     private ExistingShardsAllocator getAllocatorForShard(ShardRouting shardRouting, RoutingAllocation routingAllocation) {
         assert assertInitialized();
-        final String allocatorName = ExistingShardsAllocator.EXISTING_SHARDS_ALLOCATOR_SETTING.get(
-            routingAllocation.metadata().getIndexSafe(shardRouting.index()).getSettings()
-        );
+        String allocatorName;
+        Boolean batchModeEnabled = EXISTING_SHARDS_ALLOCATOR_BATCH_MODE.get(settings);
+        if (batchModeEnabled
+            && routingAllocation.nodes().getMinNodeVersion().onOrAfter(Version.V_2_14_0)
+            && existingShardsAllocators.size() == 2) {
+            allocatorName = ShardsBatchGatewayAllocator.ALLOCATOR_NAME;
+        } else {
+            allocatorName = ExistingShardsAllocator.EXISTING_SHARDS_ALLOCATOR_SETTING.get(
+                routingAllocation.metadata().getIndexSafe(shardRouting.index()).getSettings()
+            );
+        }
         final ExistingShardsAllocator existingShardsAllocator = existingShardsAllocators.get(allocatorName);
```

Comment:
`existingShardsAllocator` is getting referenced for start (`applyStartedShards`),failed(`applyFailedShards`) and count (`getNumberOfInFlightFetches`). Should we fix them also to use only the BatchAllocator if batch mode is enabled.

Reply:
Yes, agreed. Discussed this with @shwetathareja, for decided on creating an issue for now and make an incremental change later.

Reply:
Issue link: https://github.com/opensearch-project/OpenSearch/issues/14159